live-ingester/src/main/java/org/alfresco/hxi_connector/live_ingester/adapters/config/properties/Repository.java

```diff
@@ -27,5 +27,5 @@
 
 import jakarta.validation.constraints.NotBlank;
 
-public record Repository(@NotBlank String endpoint, @NotBlank String discoveryEndpoint, String version)
+public record Repository(@NotBlank String endpoint, String discoveryEndpoint, String versionOverride)
```

Comment:
Does this make it so that the HXI Connector can be started without defining either a `discoveryEndpoint` or `versionOverride`?

If so, we could improve the constraint validation logic to say that either A or B needs to be populated, e.g.: https://stackoverflow.com/a/66375903/11464237. This would shift the error from being a runtime occurrence to a startup occurrence, which would make it easier to spot and fix for administrators.

Reply:
Added validation in the canonical constructor. Moreover, now `DiscoveryApiClient` is annotated with `@ConditionalOnProperty("alfresco.repository.discovery-endpoint")` to not create a  bean when `alfresco.repository.discovery-endpoint` property is not set.